src/django_components/dependencies.py

```diff
@@ -734,6 +761,10 @@ def get_component_media(comp_cls_hash: str) -> Media:
     return (content, final_script_tags.encode("utf-8"), final_css_tags.encode("utf-8"))
 
 
+href_pattern = re.compile(r'href="([^"]+)"')
```

Comment:
The rest of the changes in this file is about removing dependency on BS4. For example here, to extract the URL from `<script>` or `<link>`, we'll use a regex instead of relying on BS4.